src/codegen/codegen_compatibility_visitor.cpp

```diff
@@ -73,6 +76,20 @@ std::string CodegenCompatibilityVisitor::return_error_global_var(
     return error_message_global_var.str();
 }
 
+std::string CodegenCompatibilityVisitor::return_error_param_var(
+    ast::Ast& node,
+    const std::shared_ptr<ast::Ast>& ast_node) {
+    auto param_assign = std::dynamic_pointer_cast<ast::ParamAssign>(ast_node);
+    std::stringstream error_message_global_var;
+    auto symbol = node.get_symbol_table()->lookup(param_assign->get_node_name());
+    if (!symbol->has_any_property(NmodlType::range_var) && symbol->get_write_count() > 0) {
+        error_message_global_var
+            << "\"{}\" variable found at [{}] should be defined as a RANGE variable instead of GLOBAL to enable backend transformations\n"_format(
+                   symbol->get_name(), symbol->get_token().position());
```

Comment:
`nitpick`: I am curious why description says `found at [UNKNOWN]` i.e. position information for the `T` is not available.

Reply:
I started looking at why, but it's quite hard. It comes from the lexer.

Reply:
Ok, it could be ignored here.